pupil_src/shared_modules/annotations.py

```diff
@@ -7,239 +7,333 @@
+        self._add_annotation_definition(
+            self._new_annotation_label, self._new_annotation_hotkey
+        )
+
+    def _add_annotation_definition(self, annotation_label, hotkey):
+        annotation_definition = AnnotationDefinition(
+            label=annotation_label, hotkey=hotkey
+        )
+        if annotation_definition in self._definition_to_buttons:
+            return
+        button_quickbar = self._create_button_quickbar(annotation_definition)
+        button_menu = self._create_button_menu(annotation_definition)
+        annotation_ui = self._AnnotationButtons(button_quickbar, button_menu)
+        self._definition_to_buttons[annotation_definition] = annotation_ui
+
+        self._append_button_quickbar(button_quickbar)
+        self._append_button_menu(button_menu)
+
+    def _create_button_quickbar(self, annotation_definition):
+        def make_fire(_):
+            self.fire_annotation(annotation_definition.label)
+
+        return ui.Thumb(
+            annotation_definition.label,
+            setter=make_fire,
+            getter=lambda: False,
+            label=annotation_definition.hotkey,
+            hotkey=annotation_definition.hotkey,
+        )
+
+    def _create_button_menu(self, annotation_definition):
+        def make_remove():
+            self._remove_annotation_info(annotation_definition)
+
+        label = annotation_definition.label
+        hotkey = annotation_definition.hotkey
+        return ui.Button(label=label + " <" + hotkey + ">", function=make_remove)
+
+    def _append_button_quickbar(self, button_quickbar):
+        current_buttons = self.g_pool.quickbar.elements
+        index_of_last_button = -1
+        for definition, buttons in self._definition_to_buttons.items():
+            if not buttons.quickbar:
+                continue
+            try:
+                idx = current_buttons.index(buttons.quickbar)
+            except ValueError:
+                pass
+            else:
+                index_of_last_button = max(idx, index_of_last_button)
+        if index_of_last_button != -1:
+            self.g_pool.quickbar.insert(index_of_last_button + 1, button_quickbar)
+        else:
+            self.g_pool.quickbar.append(button_quickbar)
 
-    def on_notify(self, notification):
-        if notification['subject'] == 'annotation':
-            logger.info('Received {} annotation'.format(notification['label']))
+    def _append_button_menu(self, button_menu):
+        self._annotation_list_menu.append(button_menu)
 
-    def get_init_dict(self):
-        return {'annotation_definitions': self.annotation_definitions}
+    def _remove_annotation_info(self, annotation_definition):
+        buttons = self._definition_to_buttons[annotation_definition]
+        self._remove_button_quickbar(buttons.quickbar)
+        self._remove_button_menu(buttons.menu)
+        del self._definition_to_buttons[annotation_definition]
 
+    def _remove_button_quickbar(self, button_quickbar):
```

Comment:
does it make sense to wrap these one-liners in a separate function? The one-liner is almost identical to the function names.

Reply:
I am not sure if this improves readability, but multiple usage is a good reason!